### PR TITLE
test: use testify consistently

### PR DESCRIPTION
**Pull request by kruskall**
tests are using `go-cmp`, `testify` and `t.Error`.

This standardize the assertions to use testify and forbids the other two

```release-note
-Test assertions are now standardized the to use testify
```

### CODE REVIEW

File: pkg/sbom/generator/spdx/spdx_test.go [hunk 1]
```diff
@@ -79,8 +78,7 @@ func TestReproducible(t *testing.T) {
 		require.NoError(t, err)
 		d = append(d, data)
 	}
-	diff := cmp.Diff(d[0], d[1])
-	require.Empty(t, diff)
+	require.Equal(t, d[0], d[1])
```
Comment (puerco) [inline pkg/sbom/generator/spdx/spdx_test.go:81]:
```suggestion
	diff := cmp.Diff(d[0], d[1])
	require.Empty(t, diff, fmt.Sprintf("difference in expected output %s", diff))
```

**Review by puerco — CHANGES_REQUESTED** [review]
Hey @kruskall thanks for the PR! I also prefer using testify in general but I would not go so far as to ban `google/go-cmp` completely. It has some useful methods which are useful and we can still leverage along with testify (see bellow comment).

Let's not ban outright but ensure in further reviews new tests use testify.

File: pkg/sbom/generator/spdx/spdx_test.go (hunk 1 above)
Comment (kruskall) [reply pkg/sbom/generator/spdx/spdx_test.go]: I'm probably missing something here, testify should already print the diff between "expected" and "got". I tried failing the test on purpose and the only difference I could see was `go-cmp` omitting the identical lines. Did I miss something ?

**Comment by kruskall** [comment]
Hi @puerco :wave:, Thank you for the feedback, I've reverted the last change!

**Review by puerco — APPROVED** [review]
Awesome, thank you @kruskall !!

**Pull request merged**